### ISSUE TITLE
npm install --save gh-pages failed to run on Linux virtual box environment. 

### ISSUE THREAD

**Issue by Lusimba** [npm/npm#21118]
I am using linux on virtualbox. I tried running the 'npm install --save gh-pages' on the terminal but I keep getting these warnings. 
<pre><span style="background-color:#2E3436"><font color="#D3D7CF">npm</font></span> <span style="background-color:#C4A000"><font color="#2E3436">WARN</font></span> <font color="#75507B">optional</font> Skipping failed optional dependency /chokidar/fsevents:
<span style="background-color:#2E3436"><font color="#D3D7CF">npm</font></span> <span style="background-color:#C4A000"><font color="#2E3436">WARN</font></span> <font color="#75507B">notsup</font> Not compatible with your operating system or architecture: fsevents@1.2.4
<span style="background-color:#2E3436"><font color="#D3D7CF">npm</font></span> <span style="background-color:#C4A000"><font color="#2E3436">WARN</font></span> <font color="#75507B">optional</font> Skipping failed optional dependency /react-scripts/fsevents:
<span style="background-color:#2E3436"><font color="#D3D7CF">npm</font></span> <span style="background-color:#C4A000"><font color="#2E3436">WARN</font></span> <font color="#75507B">notsup</font> Not compatible with your operating system or architecture: fsevents@1.2.4
<span style="background-color:#2E3436"><font color="#D3D7CF">npm</font></span> <span style="background-color:#C4A000"><font color="#2E3436">WARN</font></span> <font color="#75507B">optional</font> Skipping failed optional dependency /webpack-dev-server/chokidar/fsevents:
<span style="background-color:#2E3436"><font color="#D3D7CF">npm</font></span> <span style="background-color:#C4A000"><font color="#2E3436">WARN</font></span> <font color="#75507B">notsup</font> Not compatible with your operating system or architecture: fsevents@1.2.4
<span style="background-color:#2E3436"><font color="#D3D7CF">npm</font></span> <span style="background-color:#C4A000"><font color="#2E3436">WARN</font></span> ajv-keywords@3.2.0 requires a peer of ajv@^6.0.0 but none was installed</pre>
Therefore, I cannot 'npm run deploy' successfully. 
What could be the issue and how do I resolve it?

**Comment by DanielRuf** [comment]
This is due to technical issues with the npm servers, try verdaccio as local cache and `npm i` multiple times.

**Comment by kenany** [comment]
@Lusimba 

> Therefore, I cannot 'npm run deploy' successfully.

What error are you getting when you run that command?

**Comment by Lusimba** [comment]
@KenanY the error is included in the snippet captured in my issue. Some crucial dependencies fail to install and the whole process fails. Is there an alternative method of using npm with github? or, is there any other alternative?

**Comment by DanielRuf** [comment]
I would say use verdaccio as local cache and probably pnpm.

**Comment by Lusimba** [comment]
@DanielRuf I will try this these approaches. Thanks alot :-)

**Comment by kenany** [comment]
@Lusimba 

> the error is included in the snippet captured in my issue

Your post says that the snippet is the output of `npm install --save gh-pages`, but now you are saying that its the output of `npm run deploy`.

>Some crucial dependencies fail to install

`fsevents` is an optional dependency that **only works on OSX** systems. Since you are presumably not running OSX (you don't mention in your report), the dependency is skipped since it is optional. 

> the whole process fails

There are no failures in the output you have posted. It is intentional behavior for `npm` to skip optional dependencies if your system does not support them.

**Comment by DanielRuf** [comment]
Right. The warnings are ok. Is this the full content of the logfile?
